scripts/pi-hole/php/header.php

```diff
@@ -337,36 +337,29 @@ function pidofFTL()
                         } else {
                             $pistatus = null;
                         }
-                        if ($pistatus === "1") {
+                        if ($pistatus === "53") {
                             echo '<span id="status"><i class="fa fa-circle text-green-light"></i> Active</span>';
                         } elseif ($pistatus === "0") {
                             echo '<span id="status"><i class="fa fa-circle text-red"></i> Offline</span>';
                         } elseif ($pistatus === "-1") {
                             echo '<span id="status"><i class="fa fa-circle text-red"></i> DNS service not running</span>';
+                        } elseif ($pistatus === "-2" || is_null($pistatus)) {
+                            echo '<span id="status"><i class="fa fa-circle text-red"></i> Unknown</span>';
                         } else {
-                            echo '<span id="status"><i class="fa fa-circle text-orange"></i> Unknown</span>';
+                            echo '<span id="status"><i class="fa fa-circle text-orange"></i> DNS service on port '.$pistatus.'</span>';
```

Comment:
```suggestion
                            echo '<span id="status"><i class="fa fa-circle text-orange"></i> DNS service on port '.intval($pistatus).'</span>';
```
Just is case ... This avoids presentation of possibly dangerous code if something failed down the line and $pistatus contains any kind of string.

It'd be even better to have this change further up (you cannot add a suggestion to unchanged code) at
``` PHP
$pistatus = intval($pistatus[0]);
```
and fix the remainder of the code to check for integers.